### ISSUE TITLE
Crash after a few seconds since update to 3.21.3

### ISSUE THREAD

**Issue by HenningJ** [signalapp/Signal-Android#5839]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
A few seconds after opening Signal, it crashes. It always takes around 10-20 seconds, then it crashes. It doesn't matter what I do in that time inside signal. Just opening the app an waiting is enough.
I'm pretty sure it started with the last update (3.21.3).

### Steps to reproduce
- Open Signal app
- wait about 10-20 seconds
- -> chrash

### Device info
**Device:** Fairphone FP1
**Android version:** 4.2.2
**Signal version:** 3.21.3

### Link to debug log
https://gist.github.com/anonymous/b9cb6606daa4682ee99d97ceecf4d7a9
https://gist.github.com/anonymous/85a93fb440c83abb6f71b496aedb9c8e

**Comment by moxie0** [comment]
There's no crash in these logs. We'd need a debug log that covers the crash to know what's happening.

**Comment by HenningJ** [comment]
Sorry about that. 
Maybe it's not a crash. I don't get a message like "Signal has crashed". Instead, the screen turns completely black for a moment and then the app closes. 
Then I open it again and immediately send the debug log, which results in those kind of logs.

Interestingly, it just happened on the Settings screen. Instead of being returned to my android home screen, I was returned the Signal main screen. And then, a few seconds later, it happened again and I was return to the android home screen.
Hmmm...after testing it some more, it seems like it is acting more like the back button. So, I'm inside a conversation, after a few seconds -> black screen -> main Signal view, after a few seconds -> black screen -> Android home screen.

**Comment by Dyras** [comment]
Something similar to what you are describing has been happening to me as well.
https://gist.github.com/anonymous/c798e5b1eabf39e55fe4db40ba634942
I sat there in a chat just like usual, and then the screen turns black. After scratching my head for a while I get kicked back to the Signal password screen.

**Comment by moxie0** [comment]
@Dyras looks like an out of memory error. If it's reproducible, could you use the android device monitor to see where the memory is going?

**Comment by HenningJ** [comment]
hmm...I think my problem was not a Signal problem. I noticed the same thing happening with the phone app.

I had lots of duplicate contacts in my address book. One set store in the phone's internal storage and the exact same contacts synced via caldav. Now that I deleted the ones in the phone's internal storage, everything works fine again. Weird. 

So, my problem is solved, sorry for the trouble. I'll leave the ticket open for @Dyras problem.
